### ISSUE TITLE
Bactopia 2.0.0 not working as expected on macOSX

### ISSUE THREAD

**Issue by rpetit3** [bactopia/bactopia#270]
Similar issue for me. Bactopia worked well in 1.7.0 but always failed in the steps of 'ASSEMBLE_GENOME' and 'ARIBA_ANALYSIS' in 2.0.0. 
I worked in the conda environment in a macOSX machine.  

Error executing process > 'BACTOPIA:ASSEMBLE_GENOME (SRX)'

Caused by:
  Process `BACTOPIA:ASSEMBLE_GENOME (SRX)` terminated with an error exit status (127)

_Originally posted by @Jamesped in https://github.com/bactopia/bactopia/issues/268#issuecomment-1003028288_

**Comment by rpetit3** [comment]
Hi @Jamesped,

I created a new issue here. I can try to help out, but unfortunately I don't have a reliable Mac system to test with. Can you by chance share anything from the logs that might be helpful?

Thanks!

**Comment by Jamesped** [comment]
I got the same errors when going back to v1.7.1.  I decided to rebuild the datasets with `bactopia datasets' and it came out with this error message:

2022-01-08 11:51:36:root:INFO - Setting up Ariba datasets
2022-01-08 11:51:36:root:INFO - Setting up vfdb_core Ariba dataset
External command failed with exit code 1!

Command:
bash -c 'ariba prepareref -f ./datasets/ariba/vfdb_core.fa -m ./datasets/ariba/vfdb_core.tsv ./datasets/ariba/vfdb_core'

Standard error:
ERROR: I tried to get the version of bowtie2 with: "/Users/jameschen/miniconda3/envs/bactopia/bin/bowtie2 --version" and the output didn't match this regular expression: ".*bowtie2.*version (.*)$"
ERROR: I tried to get the version of nucmer with: "/Users/jameschen/miniconda3/envs/bactopia/bin/nucmer --version" and the output didn't match this regular expression: "([0-9]+\.[0-9\.]+.*$)"

Something wrong with at least one dependency. Please see the above error message(s)
ARIBA version: 2.14.6

External dependencies:
bowtie2	ERROR	/Users/jameschen/miniconda3/envs/bactopia/bin/bowtie2
cdhit	4.8.1	/Users/jameschen/miniconda3/envs/bactopia/bin/cd-hit-est
nucmer	ERROR	/Users/jameschen/miniconda3/envs/bactopia/bin/nucmer

External dependencies OK: False

Python version:
3.7.12 | packaged by conda-forge | (default, Oct 26 2021, 05:57:50)
[Clang 11.1.0 ]

Python packages:
ariba	2.14.6	/Users/jameschen/miniconda3/envs/bactopia/lib/python3.7/site-packages/ariba/__init__.py
bs4	4.10.0	/Users/jameschen/miniconda3/envs/bactopia/lib/python3.7/site-packages/bs4/__init__.py
dendropy	4.5.2	/Users/jameschen/miniconda3/envs/bactopia/lib/python3.7/site-packages/dendropy/__init__.py
pyfastaq	3.17.0	/Users/jameschen/miniconda3/envs/bactopia/lib/python3.7/site-packages/pyfastaq/__init__.py
pymummer	0.10.3	/Users/jameschen/miniconda3/envs/bactopia/lib/python3.7/site-packages/pymummer/__init__.py
pysam	0.17.0	/Users/jameschen/miniconda3/envs/bactopia/lib/python3.7/site-packages/pysam/__init__.py

Python packages OK: True

Everything looks OK: False
Some dependencies not satisfied. Cannot continue.

**Comment by rpetit3** [comment]
That's related to `tbb` take a look at this: https://github.com/bactopia/bactopia/issues/200#issuecomment-791398759

**Comment by Jamesped** [comment]
Hi @rpetit3 

I did not find too much useful information in the logs as follows. This pipeline helps a lot in my projects. Thank you for developing it and I am hopeful it can be working again. Thanks.
 

[03/87f12c] process > BACTOPIA:GATHER_SAMPLES (L09)                 [100%] 1 of 1 ✔
[0c/c85616] process > BACTOPIA:QC_READS (L09)                       [100%] 1 of 1 ✔
[f1/a097f8] process > BACTOPIA:ASSEMBLE_GENOME (L09)                [100%] 4 of 4, failed: 4, retries: 3 ✘
[-        ] process > BACTOPIA:ASSEMBLY_QC                          -
[-        ] process > BACTOPIA:ANNOTATE_GENOME                      -
[3e/047ecf] process > BACTOPIA:MINMER_SKETCH (L09)                  [100%] 1 of 1 ✔
[-        ] process > BACTOPIA:ANTIMICROBIAL_RESISTANCE             -
[3d/8e163f] process > BACTOPIA:ARIBA_ANALYSIS (L09 - vfdb_core)     [100%] 16 of 16, failed: 16, retries: 8 ✔
[16/67c62b] process > BACTOPIA:MINMER_QUERY (L09 - mash-refseq-k21) [100%] 4 of 4 ✔
[-        ] process > BACTOPIA:BLAST                                -
[-        ] process > BACTOPIA:CALL_VARIANTS                        -
[-        ] process > BACTOPIA:MAPPING_QUERY                        -
[-        ] process > BACTOPIA:SEQUENCE_TYPE                        -
[-        ] process > BACTOPIA:CUSTOM_DUMPSOFTWAREVERSIONS          -
Error executing process > 'BACTOPIA:ASSEMBLE_GENOME (L09)'

Caused by:
  Process `BACTOPIA:ASSEMBLE_GENOME (L09)` terminated with an error exit status (127)

Command executed:

  OUTDIR=results
  GENOME_SIZE=`head -n 1 L09-genome-size.txt`
  if [ "null" == "true" ]; then
      mkdir ${OUTDIR}
      gzip -cd empty.fna.gz > ${OUTDIR}/L09.fna
  elif [[ "paired-end" == "hybrid"  || "false" == "true" ]]; then
      unicycler -1 L09_R1.fastq.gz -2 L09_R2.fastq.gz -o ${OUTDIR}/ --no_correct             --min_fasta_length 500             --threads 3             --mode normal             --min_polish_size 10000             --min_component_size 1000             --min_dead_end_size 1000     --keep 1 > unicycler.stdout.txt 2> unicycler.stderr.txt
      sed -r 's/^>([0-9]+)(.*)/>L09_\1\2/' ${OUTDIR}/assembly.fasta > ${OUTDIR}/L09.fna
      mv ${OUTDIR}/assembly.gfa ${OUTDIR}/unicycler.gfa
  else
      # Shovill or Dragonflye
      shovill --R1 L09_R1.fastq.gz --R2 L09_R2.fastq.gz  --gsize ${GENOME_SIZE}             --outdir ${OUTDIR}             --force             --minlen 500             --mincov 2             --namefmt "L09_%05d"             --keepfiles             --cpus 3             --ram 7 --assembler skesa --depth 0 --noreadcorr    > shovill.stdout.txt 2> shovill.stderr.txt
      mv ${OUTDIR}/contigs.fa ${OUTDIR}/L09.fna

      # Rename Graphs
      if [ "shovill" == "shovill" ]; then
          if [ -f "${OUTDIR}/contigs.gfa" ]; then
              mv ${OUTDIR}/contigs.gfa ${OUTDIR}/skesa-unpolished.gfa
          elif [ -f "${OUTDIR}/contigs.fastg" ]; then
              mv ${OUTDIR}/contigs.fastg ${OUTDIR}/skesa-unpolished.gfa
          elif [ -f "${OUTDIR}/contigs.LastGraph" ]; then
              mv ${OUTDIR}/contigs.LastGraph ${OUTDIR}/skesa-unpolished.gfa
          fi
      fi

      if [ -f "${OUTDIR}/flye-info.txt" ]; then
          mv ${OUTDIR}/flye-info.txt ${OUTDIR}/flye.log
      fi
  fi

  # Check quality of assembly
  TOTAL_CONTIGS=`grep -c "^>" ${OUTDIR}/L09.fna || true`
  touch "total_contigs_${TOTAL_CONTIGS}"
  if [ "${TOTAL_CONTIGS}" -gt 0 ]; then
      assembly-scan ${OUTDIR}/L09.fna > ${OUTDIR}/L09.json 2> assembly-scan.stderr.txt
      TOTAL_CONTIG_SIZE=$(grep "total_contig_length" ${OUTDIR}/L09.json | sed -r 's/.*: ([0-9]+),/\1/')
      if [ "${TOTAL_CONTIG_SIZE}" -lt 100000 ]; then
          mv ${OUTDIR}/L09.fna ${OUTDIR}/L09-error.fna
          mv ${OUTDIR}/L09.json ${OUTDIR}/L09-error.json
          echo "L09 assembled size (${TOTAL_CONTIG_SIZE} bp) is less than the minimum allowed genome
                  size (100000 bp). If this is unexpected, please investigate L09 to
                  determine a cause (e.g. metagenomic, contaminants, etc...) for the poor assembly.
                  Otherwise, adjust the --min_genome_size parameter to fit your need. Further assembly
                  based analysis of L09 will be discontinued." |             sed 's/^\s*//' > L09-assembly-error.txt
      else
          # Make BLASTDB
          mkdir blastdb
          cat ${OUTDIR}/L09.fna | makeblastdb -dbtype "nucl" -title "Assembled contigs for L09" -out blastdb/L09
      fi
  else
      mv ${OUTDIR}/L09.fna ${OUTDIR}/L09-error.fna
      echo "L09 assembled successfully, but 0 contigs were formed. Please investigate
              L09 to determine a cause (e.g. metagenomic, contaminants, etc...) for this
              outcome. Further assembly-based analysis of L09 will be discontinued." |         sed 's/^\s*//' > L09-assembly-error.txt
  fi

  # Cleanup and compress
  if [ "false" == "false" ]; then
      # Remove intermediate files
      rm -rfv ${OUTDIR}/shovill.bam* ${OUTDIR}/shovill-se.bam* ${OUTDIR}/flash.extendedFrags*                  ${OUTDIR}/flash.notCombined* ${OUTDIR}/skesa.fasta* ${OUTDIR}/*.fq.gz ${OUTDIR}/00*.gfa                 ${OUTDIR}/pilon_polish* ${OUTDIR}/flye/ ${OUTDIR}/flye.fasta* ${OUTDIR}/raven/                  ${OUTDIR}/raven.fasta* ${OUTDIR}/raven.cereal ${OUTDIR}/miniasm/ ${OUTDIR}/miniasm.fasta*                 ${OUTDIR}/spades/ ${OUTDIR}/spades.fasta* ${OUTDIR}/megahit/ ${OUTDIR}/megahit.fasta*                 ${OUTDIR}/velvet.fasta* ${OUTDIR}/velvet/
  fi

  if [[ false == "false" ]]; then
      # Compress based on matched extensions
      find ${OUTDIR}/ -type f |             grep -E "\.fna$|\.fasta$|\.fa$|\.gfa$" |             xargs -I {} pigz -n --best -p 3 {}
  fi
  find ${OUTDIR} -maxdepth 1 -name "*.log" | xargs -I {} mv {} ./

  # Capture versions
  cat <<-END_VERSIONS > versions.yml
  "BACTOPIA:ASSEMBLE_GENOME":
      any2fasta:  $(echo $(any2fasta -v 2>&1) | sed 's/any2fasta //')
      assembly-scan: $(echo $(assembly-scan --version 2>&1) | sed 's/assembly-scan //')
      bwa: $(echo $(bwa 2>&1) | sed 's/^.*Version: //;s/ .*$//')
      flash: $(echo $(flash --version 2>&1) | sed 's/^.*FLASH v//;s/ .*$//')
      flye: $(echo $(flye --version))
      makeblastdb: $(echo $(makeblastdb -version 2>&1) | sed 's/^.*makeblastdb: //;s/ .*$//')
      medaka: $(echo $(medaka --version 2>&1) | sed 's/medaka //')
      megahit: $(echo $(megahit --version 2>&1) | sed 's/MEGAHIT v//')
      miniasm: $(echo $(miniasm -V))
      minimap2: $(echo $(minimap2 --version))
      nanoq: $(echo $(nanoq --version 2>&1) | sed 's/nanoq //')
      pigz: $(echo $(pigz --version 2>&1) | sed 's/pigz //')
      pilon: $(echo $(pilon --version 2>&1) | sed 's/^.*Pilon version //;s/ .*$//')
      racon: $(echo $(racon --version 2>&1) | sed 's/v//')
      rasusa: $(echo $(rasusa --version 2>&1) | sed 's/rasusa //')
      raven: $(echo $(raven --version))
      samclip: $(echo $(samclip --version 2>&1) | sed 's/samclip //')
      samtools: $(echo $(samtools --version 2>&1) |sed 's/^.*samtools //;s/ .*$//')
      shovill: $(echo $(shovill --version 2>&1) | sed 's/shovill //')
      shovill-se: $(echo $(shovill-se --version 2>&1) | sed 's/shovill-se //')
      skesa: $(echo $(skesa --version 2>&1) | sed 's/^.*SKESA //;s/ .*$//')
      spades.py: $(echo $(spades.py --version 2>&1) | sed 's/SPAdes genome assembler v//')
      velvetg: $(echo $(velvetg 2>&1) | sed 's/^.*Version //;s/ .*$//')
      velveth: $(echo $(velveth 2>&1) | sed 's/^.*Version //;s/ .*$//')
      unicycler: $(echo $(unicycler --version 2>&1) | sed 's/^.*Unicycler v//;s/ .*$//')
  END_VERSIONS

Command exit status:
  127

Command output:
  (empty)

Work dir:
  /Users/james/Desktop/raw/work/bactopia/f1/a097f82bbc703af6115daf80d8f2b0

Tip: you can replicate the issue by changing to the process work dir and entering the command `bash .command.run`

**Comment by rpetit3** [comment]
There should be some `stderr` files in `/Users/james/Desktop/raw/work/bactopia/f1/a097f82bbc703af6115daf80d8f2b0` that might have some useful info in there

**Comment by Jamesped** [comment]
I found two error files respectively related to 'ASSEMBLE_GENOME' and 'ARIBA_ANALYSIS' in the working directory. I am not sure if they were helpful.

In the assembly-scan.stderr.txt file, it showed
/Users/jameschen/Desktop/bacto/work/bactopia/1c/fa49319372c2665c11d0a073199ebd/.command.sh: line 36: assembly-scan: command not found

In the ariba.stderr.txt file, it showed

ERROR: I tried to get the version of bowtie2 with: "/Users/jameschen/miniconda3/envs/bactopia/share/bactopia-2.0.x/conda/envs/sequence_type-2.0.x/bin/bowtie2 --version" and the output didn't match this regular expression: ".*bowtie2.*version (.*)$"

Something wrong with at least one dependency. Please see the above error message(s)
ARIBA version: 2.14.6

External dependencies:
bowtie2	ERROR	/Users/jameschen/miniconda3/envs/bactopia/share/bactopia-2.0.x/conda/envs/sequence_type-2.0.x/bin/bowtie2
cdhit	4.8.1	/Users/jameschen/miniconda3/envs/bactopia/share/bactopia-2.0.x/conda/envs/sequence_type-2.0.x/bin/cd-hit-est
nucmer	3.23/scripts /Users/jameschen/CloudStation/Bioinform/cgview_comparison_tool/lib/bioperl-1.2.3 /Users/jameschen/CloudStation/Bioinform/cgview_comparison_tool/lib/perl_modules /Users/jameschen/miniconda3/envs/bactopia/lib/site_perl/5.26.2/darwin-thread-multi-2level /Users/jameschen/miniconda3/envs/bactopia/lib/site_perl/5.26.2 /Users/jameschen/miniconda3/envs/bactopia/lib/5.26.2/darwin-thread-multi-2level /Users/jameschen/miniconda3/envs/bactopia/lib/5.26.2 .) at /Users/jameschen/miniconda3/envs/bactopia/share/bactopia-2.0.x/conda/envs/sequence_type-2.0.x/bin/nucmer line 19.	/Users/jameschen/miniconda3/envs/bactopia/share/bactopia-2.0.x/conda/envs/sequence_type-2.0.x/bin/nucmer

External dependencies OK: False

Python version:
3.7.12 | packaged by conda-forge | (default, Oct 26 2021, 05:59:23) 
[Clang 11.1.0 ]

Python packages:
ariba	2.14.6	/Users/jameschen/miniconda3/envs/bactopia/share/bactopia-2.0.x/conda/envs/sequence_type-2.0.x/lib/python3.7/site-packages/ariba/__init__.py
bs4	4.10.0	/Users/jameschen/miniconda3/envs/bactopia/share/bactopia-2.0.x/conda/envs/sequence_type-2.0.x/lib/python3.7/site-packages/bs4/__init__.py
dendropy	4.5.2	/Users/jameschen/miniconda3/envs/bactopia/share/bactopia-2.0.x/conda/envs/sequence_type-2.0.x/lib/python3.7/site-packages/dendropy/__init__.py
pyfastaq	3.17.0	/Users/jameschen/miniconda3/envs/bactopia/share/bactopia-2.0.x/conda/envs/sequence_type-2.0.x/lib/python3.7/site-packages/pyfastaq/__init__.py
pymummer	0.10.3	/Users/jameschen/miniconda3/envs/bactopia/share/bactopia-2.0.x/conda/envs/sequence_type-2.0.x/lib/python3.7/site-packages/pymummer/__init__.py
pysam	0.17.0	/Users/jameschen/miniconda3/envs/bactopia/share/bactopia-2.0.x/conda/envs/sequence_type-2.0.x/lib/python3.7/site-packages/pysam/__init__.py

Python packages OK: True

Everything looks OK: False
Some dependencies not satisfied. Cannot continue.

**Comment by rpetit3** [comment]
That's related to `tbb` take a look at this: #200 (comment)

I think you'll want to do this 
```
conda activate /Users/jameschen/miniconda3/envs/bactopia/share/bactopia-2.0.x/conda/envs/sequence_type-2.0.x/
conda install -c conda-forge -c bioconda 'tbb=2020.2'
```

**Comment by jellyvanderwoude** [comment]
Mac OSX and Conda. I just installed assembly-scan independently on Conda, not sure if that is the correct fix or it's something else?

**Comment by rpetit3** [comment]
Nope! You found a bug! 

`assembly-scan` should be included here https://github.com/bactopia/bactopia/blob/master/bin/helpers/update-conda.sh#L50 Thank you for finding this, I unfortunately can test on OSX at the moment. I'll get this fixed and in the next version (hoping to release next version this week)

**Comment by rpetit3** [comment]
Happy to report this is now fixed! I've released v2.0.2 (https://github.com/bactopia/bactopia/releases/tag/v2.0.2) which includes a fix for this issue, but also many other changes! I'm going to go ahead and close this for now, please feel free to reopen!

If you have a moment, could you take a few minutes to fill out a [Bactopia Usage Survey](https://forms.gle/PTgUxEBwCsCdbQjr7)? It really helps me direct the direction of Bactopia's development! Thanks!
